plugins/steelseries/fu-steelseries-hid.c

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (C) 2022 Gaël PORTAY <gael.portay@collabora.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include "fu-steelseries-hid.h"
+
+struct _FuSteelseriesHid {
+	FuHidDevice parent_instance;
+};
+
+G_DEFINE_TYPE(FuSteelseriesHid, fu_steelseries_hid, FU_TYPE_HID_DEVICE)
+
+#define STEELSERIES_BUFFER_REPORT_SIZE 64
+#define STEELSERIES_REPORT_TIMEOUT     5000
+
+#define STEELSERIES_HID_VERSION_COMMAND	       0x90U
+#define STEELSERIES_HID_VERSION_COMMAND_OFFSET 0x00U
+#define STEELSERIES_HID_VERSION_MODE_OFFSET    0x01U
+
+static gboolean
+fu_steelseries_hid_command(FuDevice *device, guint8 *data, GError **error)
+{
+	if (!fu_hid_device_set_report(FU_HID_DEVICE(device),
+				      data[0],
+				      data,
+				      STEELSERIES_BUFFER_REPORT_SIZE,
+				      STEELSERIES_REPORT_TIMEOUT,
+				      FU_HID_DEVICE_FLAG_NONE,
+				      error)) {
+		g_prefix_error(error, "failed to do report: ");
+		return FALSE;
+	}
+
+	/* success */
+	return TRUE;
+}
+
+static gchar *
+fu_steelseries_hid_version(FuDevice *device, GError **error)
+{
+	guint8 data[STEELSERIES_BUFFER_REPORT_SIZE] = {0};
+	const guint16 cmd = 0x90U;
+	const guint8 mode = 0U; /* string */
+
+	if (!fu_common_write_uint8_safe(data,
+					sizeof(data),
+					STEELSERIES_HID_VERSION_COMMAND_OFFSET + 1,
+					cmd,
+					error))
+		return NULL;
+
+	if (!fu_common_write_uint8_safe(data,
+					sizeof(data),
+					STEELSERIES_HID_VERSION_MODE_OFFSET + 1,
+					mode,
+					error))
+		return NULL;
+
+	if (g_getenv("FWUPD_STEELSERIES_HID_VERBOSE") != NULL)
+		fu_common_dump_raw(G_LOG_DOMAIN, "Version", data, sizeof(data));
+	if (!fu_steelseries_hid_command(device, data, error))
+		return NULL;
+	if (g_getenv("FWUPD_STEELSERIES_HID_VERBOSE") != NULL)
+		fu_common_dump_raw(G_LOG_DOMAIN, "Version", data, sizeof(data));
+
+	/* success */
+	return g_strndup((const gchar *)data, sizeof(data));
+}
+
+static gboolean
+fu_steelseries_hid_probe(FuDevice *device, GError **error)
+{
+	/* FuUsbDevice->probe */
```

Comment:
FuHidDevice in the comment pls

Reply:
yes sure; lots of changes to figure out how to detect the HID device over BT.